terraform/frontend_waf.tf

```diff
@@ -0,0 +1,44 @@
+resource "aws_waf_ipset" "ipset" {
+  count = var.create_waf_acl == true ? 1 : 0
+
+  name = "keyconjurer-${terraform.workspace}-tfIPSet"
+  ip_set_descriptors {
+    type  = "IPV4"
+    value = "127.0.0.1/32"
+  }
+}
+
+resource "aws_waf_rule" "not_ip_rule" {
+  count = var.create_waf_acl == true ? 1 : 0
+
+  name        = "KeyConjurer${terraform.workspace}WafRule"
+  metric_name = "KeyConjurer${terraform.workspace}WafRule"
+
+  predicates {
+    data_id = aws_waf_ipset.ipset[0].id
+    negated = true
+    type    = "IPMatch"
+  }
+}
+
+resource "aws_waf_web_acl" "keyconjurer_waf_acl" {
+  count = var.create_waf_acl == true ? 1 : 0
+
+  name        = "KeyConjurerWAF${terraform.workspace}WebACL"
+  metric_name = "KeyConjurerWAF${terraform.workspace}WebACL"
+
+  default_action {
+    type = "ALLOW"
```

Comment:
> You may now instead pass a variable to the Terraform module, waf_acl_id, which points to an ACL; if you do not do this, a default deny-all ACL will be created and used instead.

Is this the default ACL? Should it be `BLOCK`? I have not worked with `aws_waf_web_acl` though, maybe missing something.

Reply:
Yes, this should be block for simplicity, although this WAF _does still block_ all connections.

* Default action is Allow
* but if you match `not_ip_rule` then you are blocked
* `not_ip_rule` matches all ip addresses